src/lexer.mll

```diff
@@ -219,18 +217,19 @@ let emit_verbatim input start_offset buffer =
   let t = trim_trailing_blank_lines t in
   emit input (`Verbatim t) ~start_offset
 
-let emit_code_block ~start_offset input metadata c =
-  let c = trim_trailing_blank_lines c in
+let emit_code_block ~start_offset content_offset input metadata delim terminator c has_results =
```

Comment:
The location calculations in this function are quite complicated and would deserve some comments.
Why is `content_offset` needed ?

Reply:
I've added a comment.